core/utils.py

```diff
@@ -640,6 +641,37 @@ def __init__(self):
         self.value = None
 
 
+def extract_forwarded_attachments(message) -> typing.List[typing.Tuple[str, str]]:
+    """
+    Extract attachment URLs from forwarded messages.
+
+    Parameters
+    ----------
+    message : discord.Message
+        The message to extract attachments from.
+
+    Returns
+    -------
+    List[Tuple[str, str]]
+        List of (url, filename) tuples for attachments.
+    """
+    import discord
+
+    attachments = []
+    try:
+        if getattr(message, "message_snapshots", None):
```

Comment:
Using hasattr() might be a cleaner solution for these checks. Internally python uses getattr() to make it work, but it's better for readability.
https://docs.python.org/3/library/functions.html#hasattr

Please also fix the rest of these instances (I won't be commenting on them all.)